program/src/oracle/test_oracle.c

```diff
@@ -604,24 +619,37 @@ Test( oracle, upd_aggregate ) {
   px->comp_[px->num_++].latest_ = p1;
   px->comp_[px->num_++].latest_ = p4;
   px->comp_[px->num_++].latest_ = p2;
-  upd_aggregate( px, 1001 );
+  upd_aggregate( px, 1001, 4 );
   cr_assert( px->agg_.price_ == 235 );
   cr_assert( px->agg_.conf_ == 99 );
   cr_assert( px->twap_.val_ == 124 );
   cr_assert( px->twac_.val_ == 27 );
   cr_assert( px->last_slot_ == 1001 );
   cr_assert( px->num_qt_ == 4 );
+  cr_assert( px->timestamp_ == 4 );
+  cr_assert( px->prev_slot_ == 1000 );
+  cr_assert( px->prev_price_ == 191 );
+  cr_assert( px->prev_conf_ == 74 );
+  cr_assert( px->prev_timestamp_ == 3 );
 
-  upd_aggregate( px, 1025 );
+  upd_aggregate( px, 1025, 5 );
   cr_assert( px->agg_.status_ == PC_STATUS_TRADING );
   cr_assert( px->last_slot_ == 1025 );
   cr_assert( px->num_qt_ == 4 );
+  cr_assert( px->timestamp_ == 5 );
+  cr_assert( px->prev_slot_ == 1001 );
+  cr_assert( px->prev_price_ == 235 );
+  cr_assert( px->prev_conf_ == 99 );
+  cr_assert( px->prev_timestamp_ == 4 );
 
   // check what happens when nothing publishes for a while
-  upd_aggregate( px, 1026 );
+  upd_aggregate( px, 1026, 10 );
   cr_assert( px->agg_.status_ == PC_STATUS_UNKNOWN );
   cr_assert( px->last_slot_ == 1025 );
   cr_assert( px->num_qt_ == 0 );
+  cr_assert( px->timestamp_ == 10 );
+  cr_assert( px->prev_slot_ == 1025 );
+  cr_assert( px->prev_timestamp_ == 5 );
```

Comment:
add a test case that validates that prev_* doesn't update if the agg.status is unknown prior to calling upd_aggregate